crates/data_components/src/util/on_conflict.rs

```diff
@@ -0,0 +1,232 @@
+    #[must_use]
+    pub fn build_on_conflict_statement(&self, schema: &SchemaRef) -> String {
+        match self {
+            OnConflict::DoNothingAll => "ON CONFLICT DO NOTHING".to_string(),
+            OnConflict::DoNothing(column) => {
+                format!(
+                    r#"ON CONFLICT ("{}") DO NOTHING"#,
+                    column.iter().join(r#"", ""#)
+                )
+            }
+            OnConflict::Upsert(column) => {
+                let non_constraint_columns = schema
+                    .fields()
+                    .iter()
+                    .filter(|f| !column.contains(f.name()))
+                    .map(|f| f.name().to_string())
+                    .collect::<Vec<String>>();
+                let mut update_cols = String::new();
+                for (i, col) in non_constraint_columns.iter().enumerate() {
+                    update_cols.push_str(&format!(r#""{col}" = EXCLUDED."{col}""#));
+                    if i < non_constraint_columns.len() - 1 {
+                        update_cols.push_str(", ");
+                    }
+                }
+                format!(
+                    r#"ON CONFLICT ("{}") DO UPDATE SET {update_cols}"#,
+                    column.iter().join(r#"", ""#)
+                )
+            }
+        }
+    }
+
+    #[must_use]
+    pub fn build_sea_query_on_conflict(&self, schema: &SchemaRef) -> sea_query::OnConflict {
+        match self {
+            OnConflict::DoNothingAll => {
+                let mut on_conflict = sea_query::OnConflict::new();
+                on_conflict.do_nothing();
+                on_conflict
+            }
+            OnConflict::DoNothing(column) => {
+                let mut on_conflict = sea_query::OnConflict::columns::<Vec<Alias>, Alias>(
+                    column.iter().map(Alias::new).collect(),
+                );
+                on_conflict.do_nothing();
+                on_conflict
+            }
+            OnConflict::Upsert(column) => {
+                let mut on_conflict = sea_query::OnConflict::columns::<Vec<Alias>, Alias>(
+                    column.iter().map(Alias::new).collect(),
+                );
+
+                let non_constraint_columns = schema
+                    .fields()
+                    .iter()
+                    .filter(|f| !column.contains(f.name()))
+                    .map(|f| Alias::new(f.name()))
+                    .collect::<Vec<Alias>>();
+
+                on_conflict.update_columns(non_constraint_columns);
+
+                on_conflict
+            }
+        }
+    }
+}
+
+impl Display for OnConflict {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            OnConflict::DoNothingAll => write!(f, "do_nothing_all"),
+            OnConflict::DoNothing(column) => write!(f, "do_nothing:{column}"),
+            OnConflict::Upsert(column) => write!(f, "upsert:{column}"),
+        }
+    }
+}
+
+impl TryFrom<&str> for OnConflict {
```

Comment:
These two functions are for serializing/de-serializing to strings so we can pass it as a string option in the TableProviderFactory